api/pkg/cmd/stack.go

```diff
@@ -62,36 +62,40 @@ func enrichStacklistMetadata(ctx context.Context, stacklist []string, payload mo
 	).WithTFEToken(setup.GetConfiguration().TFE.Token)
 	wg := sync.WaitGroup{}
 
-	stacks := []*model.AppStackResponse{}
-
-	for _, stackName := range stacklist {
+	stackInfos := make([]*model.AppStackResponse, len(stacklist))
+	for i, stackName := range stacklist {
 		wg.Add(1)
-		go func(stackName string) {
+		go func(id int, stackName string) {
 			defer wg.Done()
 
-			stack := &model.AppStackResponse{
-				AppMetadata: *model.NewAppMetadata(payload.AppName, payload.Environment, stackName),
-			}
-			workspace, err := workspaceRepo.GetWorkspace(ctx, fmt.Sprintf("%s-%s", payload.AppMetadata.Environment, stackName))
+			stackInfo, err := getStackInfo(ctx, payload, stackName, workspaceRepo)
```

Comment:
small goroutine refactor. I try to have as little lines in these goroutine functions as possible. it might seem silly but helps keep the concurrency pieces easy to read. Having a function that can return an error also makes the conditional logic simpler.